Lib/test/test_tools/test_pathfix.py

```diff
@@ -14,25 +14,32 @@ class TestPathfixFunctional(unittest.TestCase):
     script = os.path.join(scriptsdir, 'pathfix.py')
 
     def setUp(self):
-        self.temp_file = support.TESTFN
         self.addCleanup(support.unlink, support.TESTFN)
 
-    def pathfix(self, shebang, pathfix_flags, exitcode=0, stdout='', stderr=''):
-        with open(self.temp_file, 'w', encoding='utf8') as f:
+    def pathfix(self, shebang, pathfix_flags, exitcode=0, stdout='', stderr='',
+                directory=''):
+        if directory:
+            filename = os.path.join(directory, 'script-A_1.py')
```

Comment:
```suggestion
            filename = os.path.join(directory, 'script.py')
```

Reply:
This is on purpose. With 'script.py' the test would not have detected the issue that this pull request fixes. And I think a test case for a fix should fail without the fix and succeed afterwards. We can discuss though how we add a capital letter, a '_', a number and a '-' more nicely to the name. I am also fine if I should add a comment that explains the weird script name.

Reply:
Yeah, please add a comment like "# bpo-38347: Test filename containing uppercase, "-", "_" and digit". Otherwise, someone may remove it because it looks like a typo.

Reply:
Done in 56eaa9441a3f56420817634d910c2575ff288131.